File: 코딩테스트 입문/Javascript/배열의 길이에 따라 다른 연산하기.js
function solution(arr, n) {
    let answer = [];
    let arr_flag = arr.length%2===0 ? true : false;
    
    arr.forEach((v,i)=>{
        if(arr_flag){
            if(i%2!==0){
                answer.push(v+n);
            }
            else{
                answer.push(v);
            }
        }
        else{
            if(i%2===0){
                answer.push(v+n);
            }
            else{
                answer.push(v);
            }
        }
    })
    
    return answer;
}

File: 코딩테스트 입문/Python/배열의 길이에 따라 다른 연산하기.py
def solution(arr, n):
    flag = True if len(arr)%2==0 else False    
    answer = []
    
    # 짝수인 경우
    if flag:
        for index,value in enumerate(arr):
            if index%2!=0:
                answer.append(value+n)
            else:
                answer.append(value)
        
    # 홀수인 경우
    else:
        for index,value in enumerate(arr):
            if index%2==0:
                answer.append(value+n)
            else:
                answer.append(value)

    return answer